### ISSUE TITLE
Operation name not showing with Apollo 4 

### ISSUE THREAD

**Issue by cshrivastav007** [ChuckerTeam/chucker#1295]
## :writing_hand: Describe the bug
After upgrading to Apollo4 the chucker could not discover operation name in graphQl request.

## :wrench: Expected behavior
Correct operation name should be displayed in chucker app.

## :camera: Screenshots
<img width="435" alt="Screenshot 2024-10-08 at 10 31 49 PM" src="https://github.com/user-attachments/assets/2cee12cc-3022-4b3a-befb-40f26ef26dcf">

## :iphone: Tech info
 - Device: Samsung S23
 - OS: Android 14
 - Chucker version: 4.0.0

**Comment by cortinico** [comment]
Oh that's curious. A reproducer would help. Could you share the headers of that request you have on screen @cshrivastav007 ?

**Comment by cshrivastav007** [comment]
If you need to verify the operation name header, note that the `X-APOLLO-OPERATION-NAME` header is no longer available in Apollo version 4.

**Comment by cshrivastav007** [comment]
The last Apollo version we were using is 3.8.4 which used to have the operation name in header.
<img width="396" alt="Screenshot 2024-10-09 at 3 44 14 PM" src="https://github.com/user-attachments/assets/22ac2651-5ec5-478d-b34c-458e43e6c71d">

**Comment by cortinico** [comment]
> If you need to verify the operation name header, note that the `X-APOLLO-OPERATION-NAME` header is no longer available in Apollo version 4.

Are there no other headers where this information is available?

**Comment by cshrivastav007** [comment]
No, I believe it should be added via an interceptor by the host app (from Apollo4). For Chucker, instead of relying on the header, it would be more efficient to set the graphQlOperationName using request.operation.name()

**Comment by cortinico** [comment]
> using request.operation.name()

Chucker doesn't have access to that method though as we don't know if the user has the Apollo library installed or not. So we'll have to either inspect the body or the headers of the request

**Comment by cshrivastav007** [comment]
Ah, I see. Since it's not an Apollo Interceptor, it seems the only option is to retrieve the operation name from the decoded body content, right? 🤔

**Comment by cortinico** [comment]
> Ah, I see. Since it's not an Apollo Interceptor, it seems the only option is to retrieve the operation name from the decoded body content, right? 🤔

Yeah if there are no other headers, seems like that the preferred way

**Comment by Tamrakar182** [comment]
Thanks for this issue, I was running chucker without seeing the Graphql operation name for a while before I stumbled upon this issue. For anyone using chucker with apollo-kotlin in the near future, you can get the operation names by adding `X-APOLLO-OPERATION-NAME` header in the request. I did this by making a custom apollo interceptor and getting the operation name through `request.operation.name()` like mentioned in this thread.

Reference for my implementation:

```kotlin
class OperationNameInterceptor : ApolloInterceptor {
    override fun <D : Operation.Data> intercept(
        request: ApolloRequest<D>, chain: ApolloInterceptorChain
    ): Flow<ApolloResponse<D>> = flow {
        val opName = request.operation.name()
        val newRequest =
            request.newBuilder().addHttpHeader("X-APOLLO-OPERATION-NAME", opName).build()

        chain.proceed(newRequest).collect { emit(it) }
    }
}
```

**Comment by TWiStErRob** [comment]
Related https://www.apollographql.com/docs/kotlin/migration/4.0#non-standard-http-headers-are-not-sent-by-default
